src/Sushi.php

```diff
@@ -84,6 +84,15 @@ public static function bootSushi()
         }
     }
 
+    protected function newRelatedInstance($class)
+    {
+        return tap(new $class, function ($instance) {
+            if (!$instance->getConnectionName()) {
+                $instance->setConnection($this->getConnectionResolver()->getDefaultConnection());
```

Comment:
Should we adjust this in the framework itself? Theoretically this is an issue originating from there.. https://github.com/laravel/framework/blob/8.x/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php#L751

Reply:
We dove into this ourselves as well and it's wanted behaviour to increase support for multi tenancy. More information can be found here: https://github.com/laravel/framework/pull/16103

Might want to still consider the suggested solution since the possibility of this being fixed in the framework itself seems very limited.